pkg/schedulingtypes/interface.go

```diff
@@ -36,6 +36,8 @@ type Scheduler interface {
 	HasSynced() bool
 	Stop()
 	Reconcile(obj pkgruntime.Object, qualifiedName QualifiedName) ReconciliationStatus
+
+	RegisterPlugins(kind string, apiResource *metav1.APIResource, stopChan <-chan struct{})
```

Comment:
I think a more appropriate name for this method is `StartPlugin`.  I don't think it makes sense for the method name to be plural, since it is intended to only start a single plugin, and it will likely be necessary to add a `StopPlugin` equivalent at some point.